experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/schemaBuilder.ts

```diff
@@ -0,0 +1,127 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Assume, RestrictiveReadonlyRecord, transformObjectMap } from "../../../util";
+import { typeNameSymbol } from "../../contextuallyTyped";
+import { FieldKind } from "../../modular-schema";
+import { ImplicitFieldSchema, NormalizeField, SchemaBuilderBase } from "../../schemaBuilderBase";
+import {
+	FieldSchema,
+	InternalTypedSchemaTypes,
+	StructSchema,
+	TreeNodeSchema,
+	Unenforced,
+} from "../../typed-schema";
+import { ProxyNode, SharedTreeObject } from "./types";
+
+const factoryContent = Symbol("Node content");
+interface HasFactoryContent<T> {
+	[factoryContent]: T;
+}
+
+/**
+ * Returns the content stored on an object created by a {@link SharedTreeObjectFactory}.
+ */
+export function getFactoryContent(x: unknown): unknown | undefined {
```

Comment:
these types are not very descriptive.

Maybe x should at least be an `object`

Maybe document why this exists or what its used for, either here and/or on factoryContent or HasFactoryContent > None of these explain what a factory is, why it has content, or what the content is.

Reply:
`SharedTreeObjectFactory`, which this function points to, explains what it is. This function is internal anyway. While our docs can and will continue to improve, the documentation in this file already exceeds the status quo in this area of the API so I think it will suffice for now. We'll really nail down the perfect docs when we do a "docs pass" for MVP.

Maybe I should make `x` a `SharedTreeObject`... the question is whether this function is "a function which reads a symbol" or "a function which reads content from objects created by factories". It is documented as the latter but implemented as the former - I think you're right, I'll make it more consistent.